master/lustrefactory.py

```diff
@@ -56,6 +62,65 @@ def buildzfsCommand(props):
     return args
 
 @util.renderer
+def configureCmd(props):
+    args = ["./configure"]
+    style = props.getProperty('buildstyle')
+
+    if style == "deb" or style == "rpm":
```

Comment:
Duplicating general flow is absolutely fine.  Inventing a bunch of complicated infrastructure to avoid duplicating simple general flow is anti-KISS policy.  I've seen that many times.  "Oh these three things all need a loop, so I'll cram a bunch of complicated conditionals under one loop rather than coding three separate simple loops".

Lets say that I want to add one simple configure command line argument to one build platform.  I _could_ just go the configure line for that platform and add it under my proposed method.  Simple.  With your solution, I need to jump around to multiple areas of the code adding conditionals and properties and such.  A person needs to have a much more thorough understanding of the infrastructure to make simple changes.